src/pages/index.js

```diff
@@ -0,0 +1,349 @@
+
+api.getAppInfo()
+.then(([cards, userInfo]) => {
+  console.log("Fetched cards:", cards);
+  console.log("Number of cards:", cards.length);
+  cards.forEach((item) => {
+    const cardElement = getCardElement(item);
+    cardsList.prepend(cardElement);
+  });
+  profileNameElement.textContent = userInfo.name;
+  profileDescriptionElement.textContent = userInfo.about;
+  avatar.src = userInfo.avatar;
+})
+.catch(console.error);
+
+// function to handle closing the modal with escape key
+function handleEscapeKey(event) {
+  if (event.key === "Escape") {
+    const openedModal = document.querySelector(".modal_opened");
+    if (openedModal) {
+      closeModal(openedModal);
+    }
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown",handleEscapeKey);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown",handleEscapeKey);
+}
+
+
+//add listeners for closing the modal on overlay click
+function addOverlayClickListener(modal) {
+  modal.addEventListener("click", (event) => {
+    if (event.target === modal) {
+      closeModal(modal);
+    }
+  });
+}
+
+// select all modals
+const modals = document.querySelectorAll(".modal");
+
+avatarModalButton.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+avatarCardElement.addEventListener("submit", handleAvatarSubmit);
+
+// add listener to each modal
+modals.forEach((modal) => {
+  addOverlayClickListener(modal);
+})
+
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value
+    })
+    .then((data) => {
+      profileNameElement.textContent = data.name;
+      profileDescriptionElement.textContent = data.about;
+      avatar.src = data.avatar;
+
+      closeModal(editProfileModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+
+    });
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+
+  api
+    .addCard({
+      name: inputValues.name,
+      link: inputValues.link
+    })
+    .then((newCard) => {
+      console.log("New card added:", newCard);
+      const cardElement = getCardElement(newCard);
+      cardsList.prepend(cardElement);
+      evt.target.reset();
+      disableButton(cardSubmitButton, settings);
+      closeModal(addCardModal);
+
+    }).catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  console.log("Submit button clicked");
+  console.log("Avatar input value:", avatarLinkInput.value);
+
```

Comment:
When you change the `avatar`, the button should change the text as well to show `Saving...`

```js

  const submitBtn = evt.submitter;
  setButtonText(submitBtn, true);
```